 GC in heapDumpFinished to cleanup hprof fd

Previously the fd associated with the dump file was
not closed until the next GC. While this did not
affect AMS, external systems relying on monitoring
the close event on the file could not accurately
know when a dump operation was finished.

diff --git a/services/core/java/com/android/server/am/ActivityManagerService.java b/services/core/java/com/android/server/am/ActivityManagerService.java
index 02e693c..6bf9b6d 100644
--- a/services/core/java/com/android/server/am/ActivityManagerService.java
+++ b/services/core/java/com/android/server/am/ActivityManagerService.java
@@ -21956,6 +21956,9 @@ public void dumpHeapFinished(String path) {
             }
             if (DEBUG_PSS) Slog.d(TAG_PSS, "Dump heap finished for " + path);
             mHandler.sendEmptyMessage(POST_DUMP_HEAP_NOTIFICATION_MSG);
+
+            // Forced gc to clean up the remnant hprof fd.
+            Runtime.getRuntime().gc();
         }
     }
 
